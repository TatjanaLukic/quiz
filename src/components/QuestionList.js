import React, { useState } from "react";
import QuestionCard from "./QuestionCard";
import Loading from "./Loading";


const QuestionList = ({setCalculatedScore, questions, loading}) => {
  const [userAnswers, setUserAnswers] = useState(new Map());
  
  if(loading){
    return <Loading/>
  }

  const userAnswersUpdate = (k, v) => {
    setUserAnswers(userAnswers.set(k, v));
  };

  const onSubmitHandler = event => {
    let calculatedScore = 0;
    event.preventDefault();
    questions.forEach(el => {
      if (el.answer === userAnswers.get(`${el.id}`)) {
        calculatedScore = calculatedScore + 1;
      }
    });
    calculatedScore = calculatedScore/questions.length;
    setCalculatedScore(calculatedScore);
    window.scrollTo(0, 0);
  };

  const quizQuestions = questions.map(question => {
    return (
      <QuestionCard
        key={question.id}
        question={question}
        userAnswersUpdate={userAnswersUpdate}
      />
    );
  });

  return (
    <div className="ui form">
      {quizQuestions}
      <button
        className="ui button"
        type="submit"
        onClick={onSubmitHandler}
        style={{ marginBottom: "40px" }}
      >
        Submit
      </button>
    </div>
  );
};

export default QuestionList;
