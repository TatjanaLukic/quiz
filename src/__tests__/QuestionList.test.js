import React from "react";

import { configure, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

import QuestionList from "../components/QuestionList";

configure({ adapter: new Adapter() });

describe("<QuestionList/>", () => {
  const questions = [
    {
      id: 1,
      q: "Mock question1?",
      A: "Mock answer 1.",
      B: "Mock answer 2.",
      answer: "A",
      category: "basic",
    },
    {
      id: 2,
      q: "Mock question2?",
      A: "Mock answer 1.",
      B: "Mock answer 2.",
      answer: "A",
      category: "basic",
    },
  ];

  const setCalculatedScore = () => {};

  it("spanshot matches - not loading", () => {
    let loading = false;
    const tree = shallow(<QuestionList setCalculatedScore questions loading />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it("spanshot matches - loading", () => {
    let loading = true;
    const tree = shallow(<QuestionList setCalculatedScore questions loading />);
    expect(toJson(tree)).toMatchSnapshot();
  });

});
