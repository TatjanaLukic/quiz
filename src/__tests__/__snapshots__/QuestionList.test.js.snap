// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuestionList/> spanshot matches - loading 1`] = `<Loading />`;

exports[`<QuestionList/> spanshot matches - not loading 1`] = `<Loading />`;
